ansible/avena/roles/core/files/container-maintainer.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python3
+
+import requests   # Used to get docker-compose from server
+import os         # Used to check for existence of docker-compose files, move, and delete them
+import sys        # Used to exit the program
+import subprocess # Used to interact with docker-compose
+import socket     # Used to get hostname
+import docker     # For interacting with the docker containers
+
+# General program flow:
+# 1. Get docker-compose.yml from server use `docker-compose config` to validate
+#    a. If server docker-compose.yml file is not valid, exit
+# 2. Use `docker-compose config` to generate 'validated' version of local config
+# 3. Check if generated docker-compose.yml files differnt
+# 4. If docker-compose.yml files differ, shut down all containers and restart them
+
+# Boolean to track if a container restart is needed
+restartcontainers=False
+
+# Get current docker-compose from the server
+hostname = socket.gethostname()
+url = 'http://172.16.0.1:8006/' + hostname + '/docker-compose.yml'
+print('Querying server for docker-compose file')
+response = requests.get(url)
+
+# Ensure we got a valid response
+print('Ensuring we got a valid response')
+if response.status_code != 200:
+    print('Got a HTTP response code ', response, '. Is there a docker-compose.yml file in the right location?')
+    sys.exit()
```

Comment:
Will this leave ISOBlue dead in the field?

Reply:
No, it will use the last known good `docker-compose.yml` file it got from the server. If this is the first time it is running, there will be no docker containers running but the ISOBlue will still be accessible through SSH